File: src/main/java/com/itau/elegibilidade/inss/application/regras/Regra2.java
package com.itau.elegibilidade.inss.application.regras;

import com.itau.elegibilidade.inss.application.regras.convenio.Convenios;
import com.itau.elegibilidade.inss.application.regras.dominio.Dominios;
import com.itau.elegibilidade.inss.application.regras.modalidade.Modalidades;
import com.itau.elegibilidade.inss.entrypoint.RegraRequest;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Objects;

@Component
public class Regra2 implements Regra {

    ParametrosRegra parametros = new ParametrosRegra(
            Convenios.IPSEMG.codigo,
            Dominios.OP.codigo,
            Modalidades.PORTABILIDADE.codigo);

    @Override
    public void executar(RegraRequest request) {

        if (parametros.respeitaRequest(request))
            System.out.println("Executando regra 2");
    }
}

File: src/main/kotlin/com/itau/elegibilidade/inss/application/regras/Regra2.kt
package com.itau.elegibilidade.inss.application.regras;

import com.itau.elegibilidade.inss.application.regras.convenio.Convenios;
import com.itau.elegibilidade.inss.application.regras.dominio.Dominios;
import com.itau.elegibilidade.inss.application.regras.modalidade.Modalidades;
import com.itau.elegibilidade.inss.entrypoint.RegraRequest;
import org.springframework.stereotype.Component;

@Component
class Regra2 : Regra {

    val parametros = ParametrosRegra(
            codigoConvenio = Convenios.INSS.codigo,
            codigoDominio = Dominios.OP.codigo,
            codigoModalidade = Modalidades.REFIN.codigo)

    override fun executar(request: RegraRequest) {

        if (parametros.respeitaCondicoesEntrada(request))
            System.out.println("Executando regra 2");
    }
}
